### PR TITLE
WMCO-2.0.2: Added a release note to the WMCO 2.0.2 release

### PR DESCRIPTION
**Pull request by EricPonvelle**
This WMCO PR updates the release notes to cover the 2.0.2 bug fixes.

Version: 4.7+

Preview: https://deploy-preview-34370--osdocs.netlify.app/openshift-enterprise/latest/windows_containers/windows-containers-release-notes-2-x?utm_source=github&utm_campaign=bot_dp#wmco-2-0-2

### CODE REVIEW

File: windows_containers/windows-containers-release-notes-2-x.adoc [hunk 1]
```diff
@@ -17,6 +17,18 @@ The release notes for Red Hat OpenShift for Windows Containers tracks the develo
...
+== Release notes for Red Hat Windows Machine Config Operator 2.0.2
+
+Issued:2021-07-XX
+
+The WMCO 2.0.2 is now available with bug fixes. The components of the WMCO were released in link:https://access.redhat.com/errata/RHSA-XXXX:YYYY[RHSA-XXXX:YYYY].
+
+[id="wmco-2-0-2-bug-fixes"]
+=== Bug fixes
+
+* Previously, when you created a Windows pod, the status would remain stuck in "ContainerCreating." This bug was due to the introduction of the Windows-unsupported `os.Chown()` method. Now, when you start a Windows pod, the status updates to "Running" once the Windows pod is active. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1975553[**BZ#1975553**])
```
Comment (aravindhp) [inline windows_containers/windows-containers-release-notes-2-x.adoc:30]: The issue with the Pod stuck in `ContainerCreating` shows up only in 4.8. The reason for backporting the fix is to allow for upgrades to go through smoothly. Scenario is a 4.7.z cluster with Windows Pods and is upgraded to 4.8.z and WMCO is not upgraded to 3.0.0 it will result in the Pods being stuck in `ContainerCreating`. This is because the `BoundServiceAccountTokenVolume` is enabled by default in 4.8 and that results in Projected Volumes being attached to every Pod. In addition `RunAsUser` is added to the `SecurityContext` of the Pod and this is the root cause that triggers the Projected Volume bug with `os.Chown()` being executed for a Windows Pod and results in a failure. 

TL;DR: Customers need to upgrade to WMCO 2.0.2 before upgrading their cluster to 4.8 if they have active Windows Pods.

File: windows_containers/windows-containers-release-notes-2-x.adoc [hunk 2]
```diff
@@ -17,6 +17,18 @@ The release notes for Red Hat OpenShift for Windows Containers tracks the develo
...
+== Release notes for Red Hat Windows Machine Config Operator 2.0.2
+
+Issued:2021-07-XX
+
+The WMCO 2.0.2 is now available with bug fixes. The components of the WMCO were released in link:https://access.redhat.com/errata/RHSA-XXXX:YYYY[RHSA-XXXX:YYYY].
+
+[id="wmco-2-0-2-bug-fixes"]
+=== Bug fixes
+
+* Previously, when you created a Windows pod, the status would remain stuck in "ContainerCreating." This bug was due to the `BoundServiceAccountTokenVolume` option being enabled by default with {product-title} 4.8. This option caused projected volumes being attached to all of the pods. Finally, the `RunAsUser` option was added to the `SecurityContext` when executing the `os.Chown()` method, resulting in failure. Now, after upgrading to WMCO 2.0.2 prior to upgrading to {product-title} 4.8, he status updates to "Running" once the Windows pod is active. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1975553[**BZ#1975553**])
```
Comment (mikemckiernan) [inline windows_containers/windows-containers-release-notes-2-x.adoc:30]: Use backticks for `ContainerCreating` because it's a literal value that customers can view in the terminal.

For the second sentence--and this might be a dud idea, but consider--maybe "This bug was due to {product-title} 4.8 setting the `BoundServiceAccountTokenVolume` option by default."

Third sentence...maybe "This option caused the Operator to attach the projected volume to all the pods." (Not sure that's actually true--just my attempt at a more active voice.)

Lastly, the `Running` status should be in bacticks too instead of quotation marks--it's a literal value and all that.

Oh, change "once" to "after" or "when", from the ISG:
> Use only to mean "one time". Do not use as a conjunction to mean "after" or "when".

**Review by mikemckiernan — APPROVED** [review]
Some nits.  Let me know if I can clarify or help in any way.

File: windows_containers/windows-containers-release-notes-2-x.adoc (hunk 2 above)
Comment (aravindhp) [inline windows_containers/windows-containers-release-notes-2-x.adoc]: We should not mention `Previously, when you created a Windows pod, the status would remain stuck in "ContainerCreating."` as this cannot happen in the context of OCP 4.7 / WMCO 2.0. 
How about something like:
> `BoundServiceAccountTokenVolume` option is being enabled by default in {product-title} 4.8. This option causes projected volumes being attached to all of the pods. In addition, the `RunAsUser` option is added to the `SecurityContext` and  this combinations results in Windows Pods being stuck in `ContainerCreating`. To workaround this customers should upgrade to WMCO 2.0.2 before upgrading their cluster to 4.8.` 

We should also call out https://github.com/kubernetes/kubernetes/issues/102849 as an existing issue for Windows Pods. We should mention that we have a workaround that allows the Pods to go to running but the file permissions are not applied correctly. This applies to WMCO 1.0 also.

Comment (EricPonvelle) [reply windows_containers/windows-containers-release-notes-2-x.adoc]: Added a separate note to the Known Issues to cover this bug. Then I revised the bug fix section a little bit. PTAL

File: windows_containers/windows-containers-release-notes-2-x.adoc [hunk 3]
```diff
@@ -104,6 +115,8 @@ Windows nodes are now fully integrated with most of the monitoring capabilities
 [id="wmco-2-0-0-known-issues"]
 == Known issues
 
+* When you create a Windows pod with a projected volume with a set `RunAsUserName` option, the Windows pod will be stuck in the `ContainerCreating` state. This issue is due to file permissions not being correctly handled in the projected volume.
```
Comment (aravindhp) [inline windows_containers/windows-containers-release-notes-2-x.adoc:118]: We have a workaround for this. Rather you should say something along the lines of 

>`Windows Pods with RunAsUserName set in it's `SecurityContext` and has a Projected Volume associated with it, the file ownership permissions for the projected entities are not honored` 

as mentioned in https://github.com/kubernetes/kubernetes/issues/102849

File: windows_containers/windows-containers-release-notes-2-x.adoc [hunk 4]
```diff
@@ -104,6 +115,8 @@ Windows nodes are now fully integrated with most of the monitoring capabilities
 [id="wmco-2-0-0-known-issues"]
 == Known issues
 
+* When you create Windows pods with `RunAsUserName` set in its "SecurityContext" with a projected volume associated with these pods, the file ownership permissions for the projected entities are ignored, preventing the Windows pods from operating.
```
Comment (aravindhp) [inline windows_containers/windows-containers-release-notes-2-x.adoc:118]: It does not prevent the Windows Pods from operating. It just does not configure the file ownership permissions correctly. The Pod will go to `Running`.

**Review by aravindhp — COMMENTED** [review]
/lgtm

File: windows_containers/windows-containers-release-notes-2-x.adoc [hunk 5]
```diff
@@ -104,6 +115,8 @@ Windows nodes are now fully integrated with most of the monitoring capabilities
 [id="wmco-2-0-0-known-issues"]
 == Known issues
 
+* When you create Windows pods with `RunAsUserName` set in its "SecurityContext" with a projected volume associated with these pods, the file ownership permissions for the projected entities are ignored, resulting in incorrectly configured ownership permissions.
```
Comment (bmcelvee) [inline windows_containers/windows-containers-release-notes-2-x.adoc:118]: Feel free to ignore this comment, but this reads a bit odd to me (probably because I'm not familiar with it). Is "SecurityContext" a field `RunAsUserName` the value? If so, maybe:

When you create Windows pods with the `RunAsUserName` value set in the `SecurityContext`field with...

File: windows_containers/windows-containers-release-notes-2-x.adoc [hunk 6]
```diff
@@ -17,6 +17,17 @@ The release notes for Red Hat OpenShift for Windows Containers tracks the develo
...
+[id="wmco-2-0-2"]
+== Release notes for Red Hat Windows Machine Config Operator 2.0.2
+
+Issued:2021-07-08
+
+The WMCO 2.0.2 is now available with bug fixes. The components of the WMCO were released in link:https://access.redhat.com/errata/RHBA-2021:2671[RHBA-2021:2671].
+
+[id="wmco-2-0-2-bug-fixes"]
+=== Bug fixes
+* {product-title} 4.8 enables the `BoundServiceAccountTokenVolume` option by default. This option attaches the projected volumes to all of the pods. In addition, {product-title} 4.8 adds the `RunAsUser` option to the `SecurityContext`. This combination results in Windows pods being stuck in the `ContainerCreating` status. To workaround this issue, you should upgrade to WMCO 2.0.2 before upgrading your cluster to {product-title} 4.8. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1975553[**BZ#1975553**])
```
Comment (bmcelvee) [inline windows_containers/windows-containers-release-notes-2-x.adoc:29]: Suggested edit: 
s/workaround/work around

**Comment by bmcelvee** [comment]
Couple of suggestions, otherwise LGTM!

File: windows_containers/windows-containers-release-notes-2-x.adoc (hunk 5 above)
Comment (EricPonvelle) [reply windows_containers/windows-containers-release-notes-2-x.adoc:118]: @aravindhp Can you address this comment? I was a bit unsure and was using context clues. I believe SecurityContext is a permissions layer with `RunAsUserName` being an option.

Comment (aravindhp) [reply windows_containers/windows-containers-release-notes-2-x.adoc:118]: For context wecare talking about [Pod SecurityContext](https://docs.openshift.com/container-platform/4.7/rest_api/objects/index.html#podsecuritycontext-core-v1) here. Maybe we need to link to that?

**Comment by sjstout** [comment]
lgtm. You can go ahead and merge, especially since this if for the WINC 2.0.2 release, which went out today.

File: windows_containers/windows-containers-release-notes-2-x.adoc (hunk 5 above)
Comment (EricPonvelle) [reply windows_containers/windows-containers-release-notes-2-x.adoc:95]: Excellent! I updated the sentence where we reference the SecurityContext with a link to that Pod SecurityContext documentation.
